core/src/test/scala/unit/kafka/server/AbstractFetcherThreadTest.scala

```diff
@@ -1333,4 +1333,93 @@ class AbstractFetcherThreadTest {
 
   }
 
+  @Test
+  def testResize(): Unit = {
+
+    val topicPartition = mockTopicPartition(10, 100)
+    val threadManager: TestThreadResizeManager = new TestThreadResizeManager()
+    try {
+      threadManager.addFetcherForPartitions(topicPartition.map(_ -> InitialFetchState(None, new BrokerEndPoint(0, "localhost", 9092), 0, 1)).toMap)
+
+      threadManager.resizeThreadPool(60)
+      val threadPartitionCount = threadManager.fetcherThreadMap.map {case (brokerFetcher, thread) => s"[${brokerFetcher.fetcherId}]${thread.getName}" -> thread.partitions.size}
+      val tpsWithoutResize = mutable.Map[Int, Set[TopicPartition]]()
+      threadManager.fetcherThreadMap.foreach {
+        case (brokerFetcher, thread) => {
+          val fetcherId = brokerFetcher.fetcherId
+          val tpSet = mutable.Set[TopicPartition]()
+          thread.partitions.foreach(tp => {
+            val id = threadManager.getFetcherId(tp)
+            if (!id.equals(fetcherId)) {
+              print(s"tp: $tp with fetcherId $fetcherId which should be $id, error when resizing threads.\n")
+              tpSet += tp
+            }
+          })
+          if (!tpSet.isEmpty)
+            tpsWithoutResize += fetcherId -> tpSet
+        }
+      }
+      print(s"Current fetcher assigned partition count Map: $threadPartitionCount\nFetcher partition without resize: $tpsWithoutResize")
+      // All partitions should be redistributed to fetchers with new thread number
+      assertEquals(0, tpsWithoutResize.size)
```

Comment:
It seems that we should clean up this part. We usually don't keep print statements in tests. Overall, I think that we should verify two invariants here:
1) All the partitions must be owned by a thread; and
2) The partitions must be owned by the correct thread.

I would simplify it as follow:
```scala
val ownedPartitions = mutable.Set.empty[TopicPartition]
threadManager.fetcherThreadMap.forKeyValue { (brokerIdAndFetcherId, fetcherThread) =>
  val fetcherId = brokerIdAndFetcherId.fetchId
  fetcherThread.partitions.foreach { tp ->
    ownedPartitions += tp
    // Verify that the partition is owned by the expected fetcher thread.
    assertEquals(threadManager.getFetcherId(tp), fetcherId)
  }
}
// Verify that all partitions are owned by the fetcher threads.
assertEquals(topicPartition, ownedPartitions)
```

Note that I haven't tested this code.